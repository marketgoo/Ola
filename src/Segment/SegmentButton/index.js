import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'
import { getElementType } from '../../utils'

const SegmentButton = ({ className, onClick, as, selected, children, ...props }) => {
  const styles = cx('ola_segment-button', { 'is-selected': selected }, className)
  const ElementType = getElementType(SegmentButton, { as, ...props })
  delete props.as

  return (
    <ElementType className={styles} onClick={onClick} {...props}>
      { children }
    </ElementType>
  )
}

SegmentButton.defaultProps = {
  className: null,
  as: 'button',
  onClick: () => {},
  selected: false
}

SegmentButton.propTypes = {
  /** Extra className */
  className: PT.string,
  /** Children nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired,
  /** onClick callback function */
  onClick: PT.func,
  /** Render component as button or anchor */
  as: PT.oneOf(['button', 'a']),
  /** Indicates if button is selected or not */
  selected: PT.bool,
}

export default SegmentButton
