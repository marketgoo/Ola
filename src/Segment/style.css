.ola_segment {
  display: inline-flex;
  gap: var(--size-1);
  max-width: 100%;
  border: solid 2px var(--color-neutral-500);
  border-radius: var(--radius-xl);
  height: var(--size-5);
  box-sizing: content-box;
}
