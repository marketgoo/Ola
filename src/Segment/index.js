import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'
import SegmentButton from './SegmentButton'

const Segment = ({ className, children, ...props }) => {
  const styles = cx('ola_segment', className)

  return (
    <nav className={styles} {...props}>
      { children }
    </nav>
  )
}

Segment.Button = SegmentButton

Segment.defaultProps = {
  className: null
}

Segment.propTypes = {
  /** Extra className */
  className: PT.string,
  /** Child nodes */
  children: PT.oneOfType([
    PT.instanceOf(Segment.Button),
    PT.arrayOf(PT.instanceOf(Segment.Button)),
  ]).isRequired
}

export default Segment
