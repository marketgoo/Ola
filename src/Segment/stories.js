import React from 'react'
import Segment from './'

export default {
  title: 'Segment',
  component: Segment,
  args: {
    className: '',
  },
  argTypes: {
    children: {
      control: { disable: true }
    },
    onClick: { action: 'clicked' },
  },
  decorators: [
    (Story) => (
      <div style={{ margin: '16px' }}>
        <p style={{
          backgroundColor: 'var(--color-neutral-100)',
          padding: '16px',
          marginBottom: '32px',
        }}>Include each button as <code>&lt;Segment.Button&gt;</code>. Is <b>NOT necessary</b> to use and import <code>&lt;SegmentButton&gt;</code> directly</p>
        {Story()}
      </div>
    ),
  ],
}

export const Base = (args) => <Segment {...args}>
  <Segment.Button selected>One &lt;Segment.Button&gt;</Segment.Button>
  <Segment.Button>I am a button</Segment.Button>
  <Segment.Button as="a" href="https://www.marketgoo.com/" target="_blank">I am an anchor</Segment.Button>
</Segment>
