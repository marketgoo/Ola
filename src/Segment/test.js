import React from 'react'
import Segment from './'
import renderer from 'react-test-renderer'

it('Default Segment', () => {
  const tree = renderer
    .create(
      <Segment>
        <Segment.Button selected>One &lt;Segment.Button&gt;</Segment.Button>
        <Segment.Button>I am a button</Segment.Button>
        <Segment.Button as="a" href="https://www.marketgoo.com/" target="_blank">I am an anchor</Segment.Button>
      </Segment>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
