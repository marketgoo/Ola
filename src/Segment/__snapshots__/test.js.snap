// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Segment 1`] = `
<nav
  className="ola_segment"
>
  <button
    className="ola_segment-button is-selected"
    onClick={[Function]}
  >
    One &lt;Segment.Button&gt;
  </button>
  <button
    className="ola_segment-button"
    onClick={[Function]}
  >
    I am a button
  </button>
  <a
    className="ola_segment-button"
    href="https://www.marketgoo.com/"
    onClick={[Function]}
    target="_blank"
  >
    I am an anchor
  </a>
</nav>
`;
